src/main/java/org/broadinstitute/hellbender/tools/spark/sv/AssemblyAlignmentParser.java

```diff
@@ -0,0 +1,174 @@
+                    final ContigsCollection contigsCollection = assemblyIdAndContigsCollection._2;
+                    return contigsCollection.getContents().stream()
+                            .map(pair -> new Tuple2<>(new Tuple2<>(assemblyId, pair._1.toString()), pair._2.toString().getBytes()))
+                            .collect(Collectors.toList());
+                });
+    }
+
+    // TODO: 11/26/16 assuming 'I'/'D' cannot be the beginning or last CIGAR operator AND the following operator cannot be a clipping
+    /**
+     * Break a gapped alignment into multiple alignment regions, based on input sensitivity: i.e. if the gap(s) in the input alignment
+     * is equal to or larger than the size specified, two alignment regions will be generated.
+     * @return an iterable of size >= 1. if size==1, the returned iterable contains only the input (i.e. either no gap or hasn't reached sensitivity)
+     */
+    @VisibleForTesting
+    static Iterable<AlignmentRegion> breakGappedAlignment(final AlignmentRegion oneRegion, final int sensitivity) {
+
+        final List<CigarElement> cigarElements = oneRegion.forwardStrandCigar.getCigarElements();
+        if(cigarElements.size()==1) return new ArrayList<>( Collections.singletonList(oneRegion) );
+        final int gapCount = (int) cigarElements.stream().filter(ce -> ce.getOperator().isIndel() && ce.getLength() >= sensitivity).count();
+        if (gapCount==0) return new ArrayList<>( Collections.singletonList(oneRegion) );
+
+        // get necessary information for computing
+        final String asmId = oneRegion.assemblyId;
+        final String contigId = oneRegion.contigId;
+        final String chr = oneRegion.referenceInterval.getContig();
+        final Cigar originalForwardStrandCigar = oneRegion.forwardStrandCigar;
+        final boolean isForwardStrand = oneRegion.forwardStrand;
+        final int newMapQual = oneRegion.mapQual;
+        final int refStart = oneRegion.referenceInterval.getStart();
+        final int refEnd   = oneRegion.referenceInterval.getEnd();
+        final int contigLength = oneRegion.assembledContigLength;
+
+        // needs to accomplish three key functionality: 1. generate appropriate cigar, 2. infer reference coordinates, 3. infer contig coordinates, (cannot do yet: 4. mismatches and mapping quality)
+        final List<AlignmentRegion> result = new ArrayList<>(gapCount+1);
+
+        final List<CigarElement> cigarMemoryList = new ArrayList<>();
+        final int clippedNBases = SVVariantCallerUtils.getNumClippedBases(true, originalForwardStrandCigar);
+        final int hardClippingAtEnd = (cigarElements.get(cigarElements.size()-1).getOperator()== CigarOperator.H)? cigarElements.get(cigarElements.size()-1).getLength() : 0;
+        int refIntervalStart = refStart, contigIntervalStart = 1 + clippedNBases;
+        int gapIndex=0;
+        for(int i=0; i<cigarElements.size(); ++i){ // considers only 'M', 'I', 'D', 'S', and 'H'
```

Comment:
done for 'EQ', 'X'. Need to run on actual data to decide if 'N' or 'P' is a serious issue.